.github/workflows/deploy.py

```diff
@@ -427,6 +429,155 @@ def send_notification(url, build_url):
     requests.post(url=url, data=json.dumps(tpl))
 
 
+def get_tracer_workflows(session: requests.Session):
+    """
+    Get all avialable workflows for Tracer CI.
+    """
+    return session.get(f"https://api.github.com/repos/neonlabsorg/tracer-api/actions/workflows").json()
+
+
+def find_workflow_id_by_key(workflows, key: str, value):
+    """
+    Get workflow ID by key and value.
+    """
+    for workflow in workflows["workflows"]:
+        if workflow[key] == value:
+            return workflow["id"]
+    return None
+
+
+def get_runs_by_workflow_id(
+    session: requests.Session, workflow_id: int, created_time: str = datetime.now().strftime(DATE_FORMAT)
+):
+    """
+    Get all runs for Tracer CI and filter out them by workflow ID.
+    """
+    result = set()
+    payload = {"created": created_time}
+    resp = session.get("https://api.github.com/repos/neonlabsorg/tracer-api/actions/runs", params=payload).json()
+    if resp["total_count"] > 0:
+        for run in resp["workflow_runs"]:
+            if run["workflow_id"] == workflow_id:
+                result.add(run["id"])
+    click.echo(f"Workflow id {workflow_id} has runs: {result}.")
+    return result
+
+
+def get_runs_count(session: requests.Session, created_time: str = datetime.now().strftime(DATE_FORMAT)):
+    """
+    Get all runs for Tracer CI from particular date.
+    """
+    payload = {"created": created_time}
+    resp = session.get("https://api.github.com/repos/neonlabsorg/tracer-api/actions/runs", params=payload).json()
```

Comment:
you can add here branch name at least ( to minimalize situation when someone run the build at the same time)

Reply:
Done ✅

Reply:
There is the parameter branch for the actions/runs request
Like /actions/runs?branch=<branch_name>

Reply:
Makes sense, I'll add filtering for request in this case and remove filtering for response as it's done now 
```len([run for run in resp["workflow_runs"] if run["head_branch"] == ref])```.